### PR TITLE
Add setting Force GPU as render device

### PR DESCRIPTION
**Pull request by pdamasceno**
- Exposed ForceGPUAsRenderDevice setting under Advanced Settings/Video
- Changed the export of NvOptimusEnablement to be based on the value of ForceGPUAsRenderDevice stored in the basic.ini file

### CODE REVIEW

File: obs-studio-server/source/main.cpp [hunk 1]
```diff
@@ -53,6 +51,49 @@ extern "C" __declspec(dllexport) DWORD NvOptimusEnablement = 1;
...
+		return 1;
+	}
+
+	if (settingValue) {
+		// Disable high performance graphics rendering
+		return 0;
+	}
+
+	// Enable high performance graphics rendering
+	return 1;
```
Comment (EddyGharbi) [inline obs-studio-server/source/main.cpp]: Can we make it a bit easier to read? We could initialize the `settingValue` to true and only change the it if we successfully find the config file and find the setting in it. Then return the `settingValue` in all cases.

**Review by EddyGharbi — CHANGES_REQUESTED** [review]

File: obs-studio-server/source/main.cpp (hunk 1 above)
Comment (pdamasceno) [reply obs-studio-server/source/main.cpp]: I did the changes you asked and ended up changing the name of the setting to be able to return settingValue directly and don't make everything confusing. I believe it is better now.

File: obs-studio-server/source/nodeobs_configManager.cpp [hunk 2]
```diff
@@ -172,6 +172,11 @@ void initBasicDefault(config_t* config)
 	config_set_default_string(config, "Video", "ColorSpace", "601");
 	config_set_default_string(config, "Video", "ColorRange", "Partial");
 
+	// Set default value for EnableGPUAsRenderDevice
+	if (!config_has_user_value(config, "Video", "ForceGPUAsRenderDevice")) {
+		config_set_bool(config, "Video", "ForceGPUAsRenderDevice", true);
```
Comment (EddyGharbi) [inline obs-studio-server/source/nodeobs_configManager.cpp]: Sorry I miss this during my first review but I would use `config_set_default_bool` here in any cases, this is what it is meant for.

**Review by EddyGharbi — CHANGES_REQUESTED** [review]

File: obs-studio-server/source/nodeobs_configManager.cpp (hunk 2 above)
Comment (pdamasceno) [reply obs-studio-server/source/nodeobs_configManager.cpp]: I did the change and tested everything. It is working successfully.

**Pull request merged**